icepyx/core/visualization.py

```diff
@@ -0,0 +1,408 @@
+                    format_string = (
+                        "{product:5d}-{HH:02d}_{datetime:%Y%m%d%H%M%S}_{rgt:04d}{cycle:02d}{"
+                        "region:02d}_{release:03d}_{version:02d}.h5"
+                    )
+                else:
+                    format_string = (
+                        "{product:5d}_{datetime:%Y%m%d%H%M%S}_{rgt:04d}{cycle:02d}{"
+                        "region:02d}_{release:03d}_{version:02d}.h5"
+                    )
+                temp = intake.source.utils.reverse_format(
+                    format_string=format_string, resolved_string=fname
+                )
+
+                rgt = temp["rgt"]  # Reference Ground Track
+                cycle = temp["cycle"]  # Cycle number
+                f_date = str(temp["datetime"].date())  # Datetime
+
+                paras = [rgt, f_date, cycle, bbox_i, self.product]
+                paras_list.append(paras)
+
+        return paras_list
+
+    @backoff.on_exception(
+        backoff.expo,
+        (
+                requests.exceptions.Timeout,
+                requests.exceptions.ConnectionError,
+                requests.exceptions.ChunkedEncodingError,
+        ),
+        max_tries=5,
+    )
+    def make_request(self, base_url, payload):
+        """Make HTTP request"""
+        return requests.get(base_url, params=payload)
+
+    def request_OA_data(self, paras) -> xr.Dataset:
+        """
+        Request data from OpenAltimetry based on API:
+        https://openaltimetry.org/data/swagger-ui/#/
+
+        Parameters
+        ----------
+        paras : list
+            A single parameter list for an OpenAltimetry API data request.
+
+        Returns
+        -------
+        OA_ds : xarray.Dataset
+            An xarray Dataset containing the ICESat-2 data.
+        """
+
+        base_url = "https://openaltimetry.org/data/api/icesat2/level3a"
+        trackId, Date, cycle, bbox, product = paras
+
+        # Generate API
+        payload = {
+            "product": product.lower(),
+            "endDate": Date,
+            "minx": str(bbox[0]),
+            "miny": str(bbox[1]),
+            "maxx": str(bbox[2]),
+            "maxy": str(bbox[3]),
+            "trackId": trackId,
+            "outputFormat": "json",
+        }  # default return all six beams
+
+        # request OpenAltimetry
+        r = self.make_request(base_url, payload)
+
+        # get elevation data
+        elevation_data = r.json()
+
+        df = pd.json_normalize(data=elevation_data, record_path=["data"])
+
+        # get data we need (with the correct date)
+        try:
+
+            df_series = df.query(expr="date == @Date").iloc[0]
+            beam_data = df_series.beams
+
+        except:
+            beam_data = None
+
+        if not beam_data:
+            return
+
+        data_name = "lat_lon_elev_canopy" if product == "ATL08" else "lat_lon_elev"
+
+        sample_rate = 50 if product == "ATL06" else 20
```

Comment:
> Now I added a sample_rate variable, for ATL06 it's 1/50, for other products (segment length normally about 100 m) I use 1/20 to avoid over-sampling for low latitude regions.

This is great!